packages/widgetbook_cli/bin/main.dart

```diff
@@ -1,209 +1,15 @@
-
-  if (branchExists) {
-    baseCommit = branches
-        .firstWhere(
-          (element) => element.branchName == baseBranch,
-        )
-        .sha;
-  }
-
-  final buildPath = p.join(
-    path,
-    'build',
-    'web',
+void main(List<String> arguments, String executableName) async {
+  await flushThenExit(
+    await WidgetbookCommandRunner(executableName: executableName)
+        .run(arguments),
   );
+}
 
-  final directory = Directory(buildPath);
-  final useCases = baseBranch == null
-      ? <ChangedUseCase>[]
-      : await UseCaseParser(
-          projectPath: path,
-          baseBranch: baseBranch,
-        ).parse();
-  final themes = await ThemeParser(projectPath: path).parse();
-  final locales = await LocaleParser(projectPath: path).parse();
-  final devices = await DeviceParser(projectPath: path).parse();
-  final textScaleFactors =
-      await TextScaleFactorParser(projectPath: path).parse();
-  final file = WidgetbookZipEncoder().encode(directory);
-  if (file != null) {
-    final uploadInfo = await WidgetbookHttpClient().uploadDeployment(
-      deploymentFile: file,
-      data: DeploymentData(
-        branchName: branch,
-        repositoryName: repository,
-        commitSha: commit,
-        actor: actor,
-        apiKey: apiKey,
-        provider: gitProvider,
-      ),
-    );
-
-    if (uploadInfo != null && prNumber != null) {
-      if (gitHubToken != null) {
-        await GithubProvider(
-          apiKey: gitHubToken,
-        ).addBuildComment(
-          buildInfo: uploadInfo,
-          number: prNumber,
-        );
-      }
-    }
-
-    // If generator is not run or not properly configured
-    if (themes.isEmpty) {
-      print(
-        'HINT: Could not find generator files. '
-        'Therefore, no review has been created. '
-        'Make sure to use widgetbook_generator and '
-        'run build_runner before this CLI. '
-        'See https://docs.widgetbook.io/widgetbook-cloud/review for more '
-        'information.',
-      );
-      exit(0);
-    }
-
-    if (uploadInfo != null && baseBranch != null && baseCommit != null) {
-      await WidgetbookHttpClient().uploadReview(
-        apiKey: apiKey,
-        useCases: useCases,
-        buildId: uploadInfo['build'] as String,
-        projectId: uploadInfo['project'] as String,
-        baseBranch: baseBranch,
-        baseSha: baseCommit,
-        headBranch: branch,
-        headSha: commit,
-        themes: themes,
-        locales: locales,
-        devices: devices,
-        textScaleFactors: textScaleFactors,
-      );
-    } else {
-      print(
-        'HINT: No pull-request information available. Therefore, no review will'
-        ' be created. See docs for more information.',
-      );
-    }
-  } else {
-    print('Could not create .zip file for upload.');
-  }
+Future flushThenExit(int status) {
+  return Future.wait<void>([stdout.close(), stderr.close()])
+      .then<void>((_) => exit(status));
```

Comment:
I'm assuming that we need this code snippet, but I don't know why. 
Can you document why this method exists?